l1-contracts/src/core/libraries/FeeMath.sol

```diff
@@ -62,6 +70,46 @@ library FeeMath {
     return fakeExponential(MINIMUM_FEE_ASSET_PRICE, _numerator, FEE_ASSET_PRICE_UPDATE_FRACTION);
   }
 
+  /**
+   * @notice An approximation of the exponential function: factor * e ** (numerator / denominator)
```

Comment:
is this function taken from an existing library? if so can you add the source

Reply:
Same function described in EIP-4844. Can add a ref to it.